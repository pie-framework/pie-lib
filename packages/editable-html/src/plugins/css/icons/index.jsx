import React from 'react';
import { styled } from '@mui/material/styles';

const StyledCssIcon = styled('div')(({ theme }) => ({
  fontFamily: 'Cerebri Sans, Arial, sans-serif',
  fontSize: theme.typography.fontSize,
  fontWeight: 'bold',
  lineHeight: '14px',
  position: 'relative',
  whiteSpace: 'nowrap',
}));

const CssIcon = () => <StyledCssIcon>CSS</StyledCssIcon>;

export default CssIcon;
