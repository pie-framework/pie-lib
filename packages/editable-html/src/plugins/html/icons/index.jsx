import React from 'react';
import { styled } from '@mui/material/styles';

const StyledHtmlIcon = styled('div')(({ theme }) => ({
  fontFamily: 'Cerebri Sans, Arial, sans-serif',
  fontSize: theme.typography.fontSize,
  fontWeight: 'bold',
  lineHeight: '14px',
  position: 'relative',
  whiteSpace: 'nowrap',
}));

const HtmlModeIcon = ({ isHtmlMode }) => (
  <StyledHtmlIcon>{isHtmlMode ? 'Exit <HTML> mode' : '<HTML>'}</StyledHtmlIcon>
);

export default HtmlModeIcon;
