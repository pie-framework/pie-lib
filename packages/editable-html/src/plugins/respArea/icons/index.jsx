import React from 'react';
import PropTypes from 'prop-types';
import ChevronRight from '@mui/icons-material/ChevronRight';
import MoreVert from '@mui/icons-material/MoreVert';
import { styled } from '@mui/material/styles';

const getRotate = (direction) => {
  switch (direction) {
    case 'down':
      return 90;

    case 'up':
      return -90;

    case 'left':
      return 180;

    default:
      return 0;
  }
};

export const Chevron = (props) => {
  const { direction, style } = props;
  const rotate = getRotate(direction);

  return (
    <ChevronRight
      style={{
        transform: `rotate(${rotate}deg)`,
        ...style,
      }}
    />
  );
};

Chevron.propTypes = {
  direction: PropTypes.string,
  style: PropTypes.object,
};

export const GripIcon = ({ style }) => {
  return (
    <span style={style}>
      <MoreVert
        style={{
          margin: '0 -16px',
        }}
      />
      <MoreVert />
    </span>
  );
};

GripIcon.propTypes = {
  style: PropTypes.object,
};

const StyledToolbarIcon = styled('div')(({ theme }) => ({
  fontFamily: 'Cerebri Sans !important',
  fontSize: theme.typography.fontSize,
  fontWeight: 'bold',
  lineHeight: '14px',
  position: 'relative',
  top: '7px',
  width: '110px',
  height: '28px',
  whiteSpace: 'nowrap',
}));

export const ToolbarIcon = () => <StyledToolbarIcon>+ Response Area</StyledToolbarIcon>;
