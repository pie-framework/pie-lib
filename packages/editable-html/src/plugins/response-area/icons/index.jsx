import React from 'react';
import ChevronRight from '@material-ui/icons/ChevronRight';
import MoreVert from '@material-ui/icons/MoreVert';
import { withStyles } from '@material-ui/core/styles';

const getRotate = direction => {
  switch (direction) {
    case 'down':
      return 90;
    case 'up':
      return -90;
    case 'left':
      return 180;
    default:
      return 0;
  }
};

export const Chevron = ({ direction, style }) => {
  const rotate = getRotate(direction);

  return (
    <ChevronRight
      style={{
        transform: `rotate(${rotate}deg)`,
        ...style
      }}
    />
  );
};

export const GripIcon = ({ style }) => {
  return (
    <span style={style}>
      <MoreVert
        style={{
          margin: '0 -16px'
        }}
      />
      <MoreVert />
    </span>
  );
};

export const ToolbarIcon = withStyles({
  fontFamily: 'Cerebri Sans',
  fontSize: '14px',
  lineHeight: '14px',
  position: 'relative',
  top: '7px',
  width: '110px',
  height: '28px'
})(({ classes }) => <div className={classes.icon}>+ Response Area</div>);
