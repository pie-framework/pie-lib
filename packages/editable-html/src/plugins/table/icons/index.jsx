import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  icon: {
    fill: 'grey',
    '&:hover': {
      fill: 'black',
    },
  },
};

const SvgIcon = (Component) => {
  return withStyles(styles)((props) => (
    <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="24" height="24" viewBox="0 0 24 24">
      <Component className={props.classes.icon} />
    </svg>
  ));
};
export const AddRow = SvgIcon((props) => (
  <path
    {...props}
    d="M22,10A2,2 0 0,1 20,12H4A2,2 0 0,1 2,10V3H4V5H8V3H10V5H14V3H16V5H20V3H22V10M4,10H8V7H4V10M10,10H14V7H10V10M20,10V7H16V10H20M11,14H13V17H16V19H13V22H11V19H8V17H11V14Z"
  />
));

export const RemoveRow = SvgIcon((props) => (
  <path
    {...props}
    d="M9.41,13L12,15.59L14.59,13L16,14.41L13.41,17L16,19.59L14.59,21L12,18.41L9.41,21L8,19.59L10.59,17L8,14.41L9.41,13M22,9A2,2 0 0,1 20,11H4A2,2 0 0,1 2,9V6A2,2 0 0,1 4,4H20A2,2 0 0,1 22,6V9M4,9H8V6H4V9M10,9H14V6H10V9M16,9H20V6H16V9Z"
  />
));

export const AddColumn = SvgIcon((props) => (
  <path
    {...props}
    d="M11,2A2,2 0 0,1 13,4V20A2,2 0 0,1 11,22H2V2H11M4,10V14H11V10H4M4,16V20H11V16H4M4,4V8H11V4H4M15,11H18V8H20V11H23V13H20V16H18V13H15V11Z"
  />
));

export const RemoveColumn = SvgIcon((props) => (
  <path
    {...props}
    d="M4,2H11A2,2 0 0,1 13,4V20A2,2 0 0,1 11,22H4A2,2 0 0,1 2,20V4A2,2 0 0,1 4,2M4,10V14H11V10H4M4,16V20H11V16H4M4,4V8H11V4H4M17.59,12L15,9.41L16.41,8L19,10.59L21.59,8L23,9.41L20.41,12L23,14.59L21.59,16L19,13.41L16.41,16L15,14.59L17.59,12Z"
  />
));

export const RemoveTable = SvgIcon((props) => (
  <path
    {...props}
    d="M15.46,15.88L16.88,14.46L19,16.59L21.12,14.46L22.54,15.88L20.41,18L22.54,20.12L21.12,21.54L19,19.41L16.88,21.54L15.46,20.12L17.59,18L15.46,15.88M4,3H18A2,2 0 0,1 20,5V12.08C18.45,11.82 16.92,12.18 15.68,13H12V17H13.08C12.97,17.68 12.97,18.35 13.08,19H4A2,2 0 0,1 2,17V5A2,2 0 0,1 4,3M4,7V11H10V7H4M12,7V11H18V7H12M4,13V17H10V13H4Z"
  />
));
