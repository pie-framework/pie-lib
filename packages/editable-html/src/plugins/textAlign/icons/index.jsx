import React, { useState } from 'react';
import Collapse from '@material-ui/core/Collapse';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';

export const AlignLeft = () => (
  <svg width="20" height="20" viewBox="0 0 66 66" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M42.1875 4.75C42.1875 7.38672 39.9902 9.4375 37.5 9.4375H4.6875C2.05078 9.4375 0 7.38672 0 4.75C0 2.25977 2.05078 0.0625 4.6875 0.0625H37.5C39.9902 0.0625 42.1875 2.25977 42.1875 4.75ZM42.1875 42.25C42.1875 44.8867 39.9902 46.9375 37.5 46.9375H4.6875C2.05078 46.9375 0 44.8867 0 42.25C0 39.7598 2.05078 37.5625 4.6875 37.5625H37.5C39.9902 37.5625 42.1875 39.7598 42.1875 42.25ZM0 23.5C0 21.0098 2.05078 18.8125 4.6875 18.8125H60.9375C63.4277 18.8125 65.625 21.0098 65.625 23.5C65.625 26.1367 63.4277 28.1875 60.9375 28.1875H4.6875C2.05078 28.1875 0 26.1367 0 23.5ZM65.625 61C65.625 63.6367 63.4277 65.6875 60.9375 65.6875H4.6875C2.05078 65.6875 0 63.6367 0 61C0 58.5098 2.05078 56.3125 4.6875 56.3125H60.9375C63.4277 56.3125 65.625 58.5098 65.625 61Z"
      fill="currentColor"
    />
  </svg>
);

export const AlignRight = () => (
  <svg width="20" height="20" viewBox="0 0 66 66" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M65.625 4.75C65.625 7.38672 63.4277 9.4375 60.9375 9.4375H28.125C25.4883 9.4375 23.4375 7.38672 23.4375 4.75C23.4375 2.25977 25.4883 0.0625 28.125 0.0625H60.9375C63.4277 0.0625 65.625 2.25977 65.625 4.75ZM65.625 42.25C65.625 44.8867 63.4277 46.9375 60.9375 46.9375H28.125C25.4883 46.9375 23.4375 44.8867 23.4375 42.25C23.4375 39.7598 25.4883 37.5625 28.125 37.5625H60.9375C63.4277 37.5625 65.625 39.7598 65.625 42.25ZM0 23.5C0 21.0098 2.05078 18.8125 4.6875 18.8125H60.9375C63.4277 18.8125 65.625 21.0098 65.625 23.5C65.625 26.1367 63.4277 28.1875 60.9375 28.1875H4.6875C2.05078 28.1875 0 26.1367 0 23.5ZM65.625 61C65.625 63.6367 63.4277 65.6875 60.9375 65.6875H4.6875C2.05078 65.6875 0 63.6367 0 61C0 58.5098 2.05078 56.3125 4.6875 56.3125H60.9375C63.4277 56.3125 65.625 58.5098 65.625 61Z"
      fill="currentColor"
    />
  </svg>
);

export const AlignCenter = () => (
  <svg width="20" height="20" viewBox="0 0 66 66" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M51.5625 4.75C51.5625 7.38672 49.3652 9.4375 46.875 9.4375H18.75C16.1133 9.4375 14.0625 7.38672 14.0625 4.75C14.0625 2.25977 16.1133 0.0625 18.75 0.0625H46.875C49.3652 0.0625 51.5625 2.25977 51.5625 4.75ZM65.625 23.5C65.625 26.1367 63.4277 28.1875 60.9375 28.1875H4.6875C2.05078 28.1875 0 26.1367 0 23.5C0 21.0098 2.05078 18.8125 4.6875 18.8125H60.9375C63.4277 18.8125 65.625 21.0098 65.625 23.5ZM0 61C0 58.5098 2.05078 56.3125 4.6875 56.3125H60.9375C63.4277 56.3125 65.625 58.5098 65.625 61C65.625 63.6367 63.4277 65.6875 60.9375 65.6875H4.6875C2.05078 65.6875 0 63.6367 0 61ZM51.5625 42.25C51.5625 44.8867 49.3652 46.9375 46.875 46.9375H18.75C16.1133 46.9375 14.0625 44.8867 14.0625 42.25C14.0625 39.7598 16.1133 37.5625 18.75 37.5625H46.875C49.3652 37.5625 51.5625 39.7598 51.5625 42.25Z"
      fill="currentColor"
    />
  </svg>
);

export const AlignJustify = () => (
  <svg width="20" height="20" viewBox="0 0 66 66" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      d="M65.625 4.75C65.625 7.38672 63.4277 9.4375 60.9375 9.4375H4.6875C2.05078 9.4375 0 7.38672 0 4.75C0 2.25977 2.05078 0.0625 4.6875 0.0625H60.9375C63.4277 0.0625 65.625 2.25977 65.625 4.75ZM65.625 42.25C65.625 44.8867 63.4277 46.9375 60.9375 46.9375H4.6875C2.05078 46.9375 0 44.8867 0 42.25C0 39.7598 2.05078 37.5625 4.6875 37.5625H60.9375C63.4277 37.5625 65.625 39.7598 65.625 42.25ZM0 23.5C0 21.0098 2.05078 18.8125 4.6875 18.8125H60.9375C63.4277 18.8125 65.625 21.0098 65.625 23.5C65.625 26.1367 63.4277 28.1875 60.9375 28.1875H4.6875C2.05078 28.1875 0 26.1367 0 23.5ZM65.625 61C65.625 63.6367 63.4277 65.6875 60.9375 65.6875H4.6875C2.05078 65.6875 0 63.6367 0 61C0 58.5098 2.05078 56.3125 4.6875 56.3125H60.9375C63.4277 56.3125 65.625 58.5098 65.625 61Z"
      fill="currentColor"
    />
  </svg>
);

export default ({ editor, onChange }) => {
  const [open, setOpen] = useState(false);

  let icon;

  switch (true) {
    case editor.isActive({ textAlign: 'right' }):
      icon = <AlignRight />;
      break;
    case editor.isActive({ textAlign: 'center' }):
      icon = <AlignCenter />;
      break;
    case editor.isActive({ textAlign: 'justify' }):
      icon = <AlignJustify />;
      break;
    default:
      icon = <AlignLeft />;
      break;
  }

  const applyAlignment = (event) => {
    const alignType = event.target?.closest('div')?.getAttribute('value');

    if (alignType) {
      editor.commands.setTextAlign(alignType);
    }

    setOpen(false);
  };

  const onMouseDown = (event) => {
    event.preventDefault();
    event.stopPropagation();
    setOpen(!open);
  };

  return (
    <div>
      <div style={{ display: 'flex', alignItems: 'center' }} onClick={onMouseDown}>
        {icon}
        <span style={{ marginLeft: '5px', fontSize: '8px' }}>&#9660;</span>
      </div>
      <Collapse in={open} timeout="auto" unmountOnExit style={{ position: 'absolute' }}>
        <List
          component="div"
          disablePadding
          style={{
            background: '#fff',
            display: 'flex',
            flexDirection: 'row',
            padding: 0,
          }}
        >
          <ListItem button type="submit" value="left" aria-label="Align text left" onClick={applyAlignment}>
            <AlignLeft />
          </ListItem>

          <ListItem button type="submit" value="center" aria-label="Align text center" onClick={applyAlignment}>
            <AlignCenter />
          </ListItem>

          <ListItem button type="submit" value="right" aria-label="Align text right" onClick={applyAlignment}>
            <AlignRight />
          </ListItem>

          <ListItem button type="submit" value="justify" aria-label="Justify text" onClick={applyAlignment}>
            <AlignJustify />
          </ListItem>
        </List>
      </Collapse>
    </div>
  );
};
