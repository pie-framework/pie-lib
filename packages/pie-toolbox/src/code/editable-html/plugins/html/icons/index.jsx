import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = (theme) => ({
  icon: {
    fontFamily: 'Cerebri Sans, Arial, sans-serif',
    fontSize: theme.typography.fontSize,
    fontWeight: 'bold',
    lineHeight: '14px',
    position: 'relative',
    whiteSpace: 'nowrap',
  },
});

const HtmlModeIcon = ({ classes, isHtmlMode }) => (
  <div className={classes.icon}>{isHtmlMode ? 'Exit <HTML> mode' : '<HTML>'}</div>
);

export default withStyles(styles)(HtmlModeIcon);
