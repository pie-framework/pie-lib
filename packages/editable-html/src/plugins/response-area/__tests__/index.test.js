import React from 'react';
import { shallow } from 'enzyme';
import { render } from 'react-dom';
import { Data, Text } from 'slate';
import { ToolbarIcon } from '../icons';
import RespAreaToolbar, { serialization } from '../index';

jest.useFakeTimers();

let mockCloneFragment, mockFindDOMNode, mockGetEventTransfer;

jest.mock('slate-react', () => {
  mockCloneFragment = jest.fn();
  mockFindDOMNode = jest.fn().mockReturnValue({
    getBoundingClientRect: jest.fn().mockReturnValue({
      height: 200,
      top: 500
    }),
    closest: jest.fn().mockReturnValue({
      getBoundingClientRect: jest.fn().mockReturnValue({
        top: 100
      })
    })
  });
  mockGetEventTransfer = jest.fn();

  return {
    cloneFragment: mockCloneFragment,
    findDOMNode: mockFindDOMNode,
    getEventTransfer: mockGetEventTransfer
  };
});

describe('response-area', () => {
  const respArea = RespAreaToolbar({ type: 'explicit-constructed-response', options: {} });

  describe('name', () => {
    it('should have the name', () => {
      expect(respArea.name).toEqual('response_area');
    });
  });
  describe('filterPlugins', () => {
    it('should return an appropriate value for a node type', () => {
      expect(respArea.filterPlugins({ type: 'explicit_constructed_response' })).toEqual([]);
      expect(
        respArea.filterPlugins({ type: 'drag_in_the_blank' }, [
          { name: 'image' },
          { name: 'response_area' }
        ])
      ).toEqual([{ name: 'image' }]);
    });
  });
  describe('pluginStyles', () => {
    let document = jest.spyOn(window, 'getComputedStyle');

    it('should not return anything if the plugin is not math', () => {
      document.mockReturnValue({ width: '500px' });
      expect(respArea.pluginStyles({}, { name: 'inline_dropdown' })).toEqual(undefined);
      expect(respArea.pluginStyles({ key: '0' }, { name: 'math' })).toMatchObject({
        position: 'absolute',
        top: '600px',
        width: '500px'
      });
    });
  });
  describe('stopReset', () => {
    it('should return true', () => {
      expect(respArea.stopReset()).toEqual(true);
    });
  });
  describe('renderNode', () => {
    let attributes = { key: '0' },
      n,
      node;
    beforeEach(() => {
      n = { key: '0' };
      node = { key: '0' };
    });

    describe('item_builder', () => {
      it('should render an item_builder', () => {
        const el = respArea.renderNode({
          attributes,
          n,
          node: { key: '0', type: 'item_builder' }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('response_menu_item', () => {
      it('should render children', () => {
        const el = respArea.renderNode({
          children: <div>A</div>,
          node: { key: '0', type: 'response_menu_item' }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe('menu_item', () => {
      it('should render an menu_item', () => {
        const el = respArea.renderNode({
          attributes,
          n,
          node: {
            key: '0',
            type: 'menu_item',
            data: Data.create({})
          }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe('explicit_constructed_response', () => {
      it('should render an explicit_constructed_response', () => {
        const el = respArea.renderNode({
          attributes,
          n,
          node: {
            key: '0',
            type: 'explicit_constructed_response',
            data: Data.create({})
          }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe('drag_in_the_blank', () => {
      it('should render an drag_in_the_blank', () => {
        const el = respArea.renderNode({
          attributes,
          n,
          node: {
            key: '0',
            type: 'drag_in_the_blank',
            data: Data.create({})
          }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe('inline_dropdown', () => {
      it('should render an inline_dropdown', () => {
        const el = respArea.renderNode({
          attributes,
          n,
          children: [],
          node: {
            key: '0',
            type: 'inline_dropdown',
            data: Data.create({})
          }
        });
        const wrapper = shallow(el);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
  describe('normalizeNode', () => {
    it('should return undefined when node is not of correct type', () => {
      expect(
        respArea.normalizeNode({
          type: 'foo'
        })
      ).toEqual(undefined);
    });
    it('should return undefined when node is normalized already', () => {
      const val = respArea.normalizeNode({
        type: 'inline_dropdown',
        data: Data.create({}),
        findDescendant: jest.fn().mockReturnValue({
          getText: jest.fn().mockReturnValue('')
        })
      });
      expect(val).toEqual(undefined);
    });
  });
});
describe('serialization', () => {
  describe('deserialize', () => {
    let next;

    beforeEach(() => {
      next = jest.fn().mockReturnValue([]);
    });

    describe('inline_dropdown', () => {
      let el;
      let out;
      beforeEach(() => {
        el = {
          childNodes: [],
          dataset: {
            correctId: '0',
            type: 'inline_dropdown',
            inTable: true
          }
        };
      });
      it('returns a inline_dropdown element with 2 default options if no menu items', () => {
        const createDefText = index => `Default Option ${index}`;

        out = serialization.deserialize(el, next);

        expect(out).toMatchObject({
          object: 'inline',
          type: 'inline_dropdown',
          data: {
            selected: 0,
            inTable: true
          },
          nodes: [
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 1,
                value: createDefText('1'),
                isDefault: true
              }
            },
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 2,
                value: createDefText('2'),
                isDefault: true
              }
            }
          ]
        });
      });

      it('returns a inline_dropdown element with 1 default option if it has 1 menu item', () => {
        const createDefText = index => `Default Option ${index}`;

        next = jest.fn(arr => arr);

        el.childNodes = [
          {
            object: 'inline',
            type: 'menu_item',
            data: {
              id: 1,
              value: 'Foo Bar'
            }
          }
        ];
        out = serialization.deserialize(el, next);

        expect(out).toMatchObject({
          object: 'inline',
          type: 'inline_dropdown',
          data: {
            selected: 0,
            inTable: true
          },
          nodes: [
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 1,
                value: 'Foo Bar'
              }
            },
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 2,
                value: createDefText('2'),
                isDefault: true
              }
            }
          ]
        });
      });

      it('returns a inline_dropdown element with the computed nodes if at least 2 menu items', () => {
        next = jest.fn(arr => arr);

        el.childNodes = [
          {
            object: 'inline',
            type: 'menu_item',
            data: {
              id: 1,
              value: 'Foo Bar'
            }
          },
          {
            object: 'inline',
            type: 'menu_item',
            data: {
              id: 2,
              value: 'Bar Foo'
            }
          }
        ];
        out = serialization.deserialize(el, next);

        expect(out).toMatchObject({
          object: 'inline',
          type: 'inline_dropdown',
          data: {
            selected: 0,
            inTable: true
          },
          nodes: [
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 1,
                value: 'Foo Bar'
              }
            },
            {
              object: 'inline',
              type: 'menu_item',
              data: {
                id: 2,
                value: 'Bar Foo'
              }
            }
          ]
        });
      });
    });

    describe('menu_item', () => {
      let el;
      let out;

      beforeEach(() => {
        el = {
          childNodes: [],
          dataset: {
            id: '0',
            type: 'menu_item',
            inTable: true
          }
        };
      });

      it('should return a menu_item', () => {
        el.textContent = 'Foo Bar';
        next = jest.fn(arr => arr);
        out = serialization.deserialize(el, next);
        expect(out).toMatchObject({
          object: 'inline',
          type: 'menu_item',
          data: {
            id: '0',
            value: 'Foo Bar'
          }
        });
      });

      it('should return a menu_item with isDefault set to true', () => {
        el.textContent = 'Foo Bar';
        el.dataset.isDefault = true;
        next = jest.fn(arr => arr);
        out = serialization.deserialize(el, next);

        expect(out).toMatchObject({
          object: 'inline',
          type: 'menu_item',
          data: {
            id: '0',
            value: 'Foo Bar',
            isDefault: true
          }
        });
      });
    });

    describe('explicit_constructed_response', () => {
      let el;
      let out;

      beforeEach(() => {
        el = {
          childNodes: [],
          dataset: {
            type: 'explicit_constructed_response'
          }
        };
      });

      it('should return a explicit_constructed_response', () => {
        next = jest.fn(arr => arr);
        out = serialization.deserialize(el, next);
        expect(out).toMatchObject({
          object: 'inline',
          type: 'explicit_constructed_response',
          data: {
            inTable: undefined
          }
        });
      });

      it('should return a explicit_constructed_response with isDefault set to true', () => {
        el.dataset.inTable = true;
        next = jest.fn(arr => arr);
        out = serialization.deserialize(el, next);

        expect(out).toMatchObject({
          object: 'inline',
          type: 'explicit_constructed_response',
          data: {
            inTable: true
          }
        });
      });
    });

    describe('drag_in_the_blank', () => {
      let el;
      let out;

      beforeEach(() => {
        el = {
          childNodes: [],
          dataset: {
            type: 'drag_in_the_blank'
          }
        };
      });

      it('should return a drag_in_the_blank', () => {
        el.dataset = {
          ...el.dataset,
          index: 0,
          id: '0',
          value: 'Foo Bar',
          inTable: true
        };
        next = jest.fn(arr => arr);
        out = serialization.deserialize(el, next);
        expect(out).toMatchObject({
          object: 'inline',
          type: 'drag_in_the_blank',
          data: {
            index: 0,
            id: '0',
            value: 'Foo Bar',
            inTable: true
          }
        });
      });
    });
  });

  describe('serialize', () => {
    it('should return a inline_dropdown', () => {
      const el = serialization.serialize(
        {
          object: 'inline',
          type: 'inline_dropdown',
          nodes: [],
          data: Data.create({
            selected: '0',
            inTable: true
          })
        },
        'A'
      );
      expect(el).toEqual(
        <span data-type="inline_dropdown" data-correct-id="0" data-in-table={true}>
          A
        </span>
      );
    });
    it('should return a explicit_constructed_response', () => {
      const el = serialization.serialize(
        {
          object: 'inline',
          type: 'explicit_constructed_response',
          nodes: [],
          data: Data.create({
            inTable: true
          })
        },
        'A'
      );
      expect(el).toEqual(
        <span data-type="explicit_constructed_response" data-in-table={true}>
          A
        </span>
      );
    });

    it('should return a drag_in_the_blank', () => {
      const el = serialization.serialize(
        {
          object: 'inline',
          type: 'drag_in_the_blank',
          nodes: [],
          data: Data.create({
            index: 0,
            id: '0',
            value: 'Foo Bar',
            inTable: true
          })
        },
        'A'
      );

      expect(el).toEqual(
        <span
          data-type="drag_in_the_blank"
          data-index={0}
          data-id="0"
          data-value="Foo Bar"
          data-in-table={true}
        />
      );
    });

    it('should return null', () => {
      const el = serialization.serialize({ object: 'inline', type: 'item_builder' });

      expect(el).toEqual(null);
    });

    it('should return null', () => {
      const el = serialization.serialize({ object: 'inline', type: 'response_menu_item' });

      expect(el).toEqual(null);
    });
    it('should return a menu_item', () => {
      const el = serialization.serialize(
        {
          object: 'inline',
          type: 'menu_item',
          nodes: [],
          data: Data.create({
            id: '0',
            isDefault: true
          })
        },
        'A'
      );
      expect(el).toEqual(
        <span data-type="menu_item" data-id="0" data-is-default={true}>
          A
        </span>
      );
    });
  });
});
