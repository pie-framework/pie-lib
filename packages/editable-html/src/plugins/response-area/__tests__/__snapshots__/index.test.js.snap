// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`response-area renderNode drag_in_the_blank should render an drag_in_the_blank 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "margin": "0 10px",
      "minHeight": "50px",
      "minWidth": "178px",
      "position": "relative",
    }
  }
>
  <DragSource(DropTarget(WithStyles(Component)))
    dragKey="0"
    n={
      Object {
        "data": Immutable.Map {},
        "key": "0",
        "type": "drag_in_the_blank",
      }
    }
    onChange={[Function]}
    removeResponse={[Function]}
    targetId="0"
    value={Object {}}
  />
</span>
`;

exports[`response-area renderNode explicit_constructed_response should render an explicit_constructed_response 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "margin": "0 10px",
      "minHeight": "50px",
      "minWidth": "178px",
      "position": "relative",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": 0,
        "cursor": "default",
        "display": "none",
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
  />
  <div
    contentEditable={true}
    data-key="0"
    onMouseDown={[Function]}
    style={
      Object {
        "background": "#FFF",
        "border": "1px solid #C0C3CF",
        "borderRadius": "3px",
        "boxSizing": "border-box",
        "display": "inline-flex",
        "height": "36px",
        "minHeight": "36px",
        "minWidth": "178px",
        "overflow": "hidden",
        "padding": "8px",
        "position": "relative",
      }
    }
    suppressContentEditableWarning={true}
  />
</span>
`;

exports[`response-area renderNode inline_dropdown should render an inline_dropdown 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "height": "50px",
      "margin": "0 10px",
      "position": "relative",
      "top": "10px",
    }
  }
>
  <div
    contentEditable={false}
    onClick={[Function]}
    style={
      Object {
        "background": "#FFF",
        "border": "1px solid #C0C3CF",
        "borderRadius": "3px",
        "bottom": "10px",
        "boxSizing": "border-box",
        "display": "inline-flex",
        "height": "36px",
        "minWidth": "178px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "background": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <div
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
          "padding": "8px 25px 8px 8px",
        }
      }
    >
      Â 
    </div>
    <Chevron
      direction="down"
      style={
        Object {
          "position": "absolute",
          "right": "5px",
          "top": "5px",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": 0,
        "cursor": "default",
        "display": "none",
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#fff",
        "border": "1px solid #E0E1E6",
        "borderRadius": "3px",
        "boxShadow": "0px 0px 5px rgba(126, 132, 148, 0.3)",
        "boxSizing": "border-box",
        "display": "block",
        "maxHeight": "400px",
        "minWidth": "178px",
        "overflow": "scroll",
        "position": "absolute",
        "top": "30px",
        "visibility": "hidden",
        "zIndex": -1,
      }
    }
  />
</span>
`;

exports[`response-area renderNode item_builder should render an item_builder 1`] = `
<span
  contentEditable={true}
  key="0"
  style={
    Object {
      "alignItems": "center",
      "background": "#E0E1E6",
      "borderRadius": "2px 2px 0px 0px",
      "boxSizing": "border-box",
      "cursor": "initial",
      "display": "inline-flex",
      "justifyContent": "center",
      "minHeight": "52px",
      "minWidth": "178px",
      "padding": "10px",
      "width": "100%",
    }
  }
  suppressContentEditableWarning={true}
>
  <span
    data-key="0"
    onMouseDown={[Function]}
    style={
      Object {
        "background": "#fff",
        "border": "2px solid #89B7F4",
        "borderRadius": "3px",
        "boxSizing": "border-box",
        "display": "inline-block",
        "minHeight": "36px",
        "minWidth": "160px",
        "padding": "10px 25px 10px 10px",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <i
      contentEditable={false}
      onMouseDown={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "fontStyle": "normal",
          "position": "absolute",
          "right": "8px",
          "top": "8px",
        }
      }
    >
      +
    </i>
  </span>
</span>
`;

exports[`response-area renderNode menu_item should render an menu_item 1`] = `
<span
  contentEditable={false}
  data-node-key="0"
  onMouseDown={[Function]}
  style={
    Object {
      "background": "#fff",
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "block",
      "lineHeight": "30px",
      "margin": "0px 0px -20px 0px",
      "minHeight": "45px",
      "minWidth": "178px",
      "padding": "10px 25px 10px 10px",
      "position": "relative",
    }
  }
>
  <i
    data-key="0"
    onMouseDown={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "20px",
        "fontStyle": "normal",
        "height": "50px",
        "lineHeight": "40px",
        "position": "absolute",
        "right": "0",
        "top": "0",
        "width": "25px",
        "zIndex": 3,
      }
    }
  >
    x
  </i>
  <div
    style={
      Object {
        "background": "transparent",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
  />
</span>
`;

exports[`response-area renderNode response_menu_item should render children 1`] = `
<div>
  A
</div>
`;
